src/styles/index.scss

```diff
@@ -1,3 +1,78 @@
-body {
-  margin: 0;
+@use 'sass:list';
+@import 'variables';
+
+.page {
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+  &__body {
+    margin: 0;
+    height: 100vh;
+
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+
+  // Final column width including the total width of days and the gaps between them
+  width: calc((#{$day-size} * #{$columns}) + (#{$gap} * #{$columns - 1}));
+  background-color: #fff;
+
+  &__day {
+    width: $day-size;
+    height: $day-size;
+    background-color: #eee;
+    display: flex;
+    border: 1px solid #000;
+    box-sizing: border-box;
+    transition:
+      background-color 0.5s,
+      transform 0.5s;
+
+    &::before {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      margin: auto;
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+      transition:
+        background-color 0.5s,
+        transform 0.5s;
+    }
+  }
+
+  $days: mon, tue, wed, thu, fri, sat, sun;
+
+  @each $day in $days {
+    $index: list.index($days, $day) - 1;
```

Comment:
The `list.index` function is used here, which is correct, but ensure that the `sass:list` module is correctly imported at the beginning of the file, as it is necessary for this function to work.